src/main/java/com/manolito/dashflow/service/application/IssuesService.java

```
@@ -21,19 +22,22 @@ public class IssuesService {
      * @param priority the priority level of the issues to filter
-     * @return a map where the key is the issue type ("Bug", "Enhancement", "Question") and the value is the count
+     * @return a list of IssueCount objects containing issue type and count
      * @throws NoSuchElementException if no issues are found for the given criteria
      */
-    public Map<String, Integer> getIssueCountsByProjectSeverityAndPriority(
-            int projectId, IssueSeverity severity, IssuePriority priority) {
+    public List<IssueCountDto> getIssueCountsByProjectSeverityAndPriority(
+            String projectId, IssueSeverity severity, IssuePriority priority) {
 
-        Map<String, Integer> result = new HashMap<>();
+        List<IssueCountDto> result = new ArrayList<>();
 
-        result.put("Bug", issuesDataWarehouseRepository.getIssueCountByType(projectId, severity.getValue(), priority.getValue(), "Bug").orElse(0));
-        result.put("Enhancement", issuesDataWarehouseRepository.getIssueCountByType(projectId, severity.getValue(), priority.getValue(), "Enhancement").orElse(0));
-        result.put("Question", issuesDataWarehouseRepository.getIssueCountByType(projectId, severity.getValue(), priority.getValue(), "Question").orElse(0));
+        result.add(new IssueCountDto("Bug",
+                issuesDataWarehouseRepository.getIssueCountByType(projectId, severity.getValue(), priority.getValue(), "Bug").orElse(0)));
+        result.add(new IssueCountDto("Enhancement",
+                issuesDataWarehouseRepository.getIssueCountByType(projectId, severity.getValue(), priority.getValue(), "Enhancement").orElse(0)));
+        result.add(new IssueCountDto("Question",
+                issuesDataWarehouseRepository.getIssueCountByType(projectId, severity.getValue(), priority.getValue(), "Question").orElse(0)));
 
-        if (result.values().stream().allMatch(count -> count == 0)) {
+        if (result.stream().allMatch(issueCount -> issueCount.getCount() == 0)) {
```

You don't need to return an exception here, just return empty